rotkehlchen/tests/integration/test_blockchain.py

```diff
@@ -104,11 +106,20 @@ def mock_add_defi_balances_to_token_and_totals():
             blockchain=SupportedBlockchain.ETHEREUM,
             accounts=[addr1, addr2],
         )
+        ethtokens = EthTokens(database=blockchain.database, ethereum=blockchain.ethereum)
+        with ethtokens.db.user_write() as write_cursor:
+            ethtokens.detect_tokens(
+                write_cursor=write_cursor,
+                only_cache=False,
+                accounts=[addr1, addr2],
+            )
+
     assert addr1 in blockchain.accounts.eth
 
     with etherscan_patch, ethtokens_max_chunks_patch, defi_balances_mock, add_defi_mock, beaconchain_patch:  # noqa: E501
+        blockchain.query_ethereum_balances(ignore_cache=True)
```

Comment:
With this new approach of splitting token detection I am not so sure this makes much sense anymore though. So what are you doing here?

Detecting tokens at like 111? And then querying balances once without cache and then 5 times with cache? Why did you add the one time without cache in the beginning?